perf(linter/no-unescaped-entities): add fast path to check if char should be replaced

camchenry opened the pull request:
Flamegraph shows a lot of time spent hashing here, so I added a faster check to see if it is a char that we need to replace.

camchenry commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6594?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6594** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6594?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#6590** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6590?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camchenry commented:
+2% on `RadixUIAdoptionSection.jsx` and +1% on `checker.ts`, seems worth it!

<img width="674" alt="image" src="https://github.com/user-attachments/assets/62567236-643a-4f47-925b-f3ff6896e558">

DonIsaac commented:
### Merge activity

* **Oct 16, 3:58 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
perf(linter/no-unescaped-entities): add fast path to check if char should be replaced (#6594)

Flamegraph shows a lot of time spent hashing here, so I added a faster check to see if it is a char that we need to replace.
